acceptance/tests/resource/service/launchd_provider.rb

```diff
@@ -0,0 +1,82 @@
+test_name 'Mac OS X launchd Provider Testing'
+ 
+confine :to, {:platform => /osx/}, agents
+ 
+sloth_daemon_script = <<SCRIPT
+#!/usr/bin/env sh
+while true; do sleep 1; done
+SCRIPT
+
+svc = 'com.puppetlabs.sloth'
+launchd_script_path = "/Library/LaunchDaemons/#{svc}.plist"
+
+def assert_service_status_on(host, service, status)
+  ensure_status = "ensure => '#{status[:ensure]}'" if status[:ensure]
+  enable_status = "enable => '#{status[:enable]}'" if status[:enable]
+
+  on host, puppet_resource('service', service) do
+    assert_match(/#{service}.+#{ensure_status}.+#{enable_status}/, stdout, "Service status does not match expectation #{status}")
+  end
+end
+
+def ensure_service_on(host, service, status)
+  ensure_status = "ensure => '#{status[:ensure]}'," if status[:ensure]
+  enable_status = "enable => '#{status[:enable]}'," if status[:enable]
+
+  apply_manifest_on host, "service { '#{service}': #{ensure_status} #{enable_status} }" do
+    assert_match(/Service\[#{service}\]\/ensure: ensure changed '\w+' to '#{status[:ensure]}'/, stdout,
+                 'Service status change failed') if status[:ensure]
+    assert_match(/Service\[#{service}\]\/enable: enable changed '\w+' to '#{status[:enable]}'/, stdout,
+                 'Service enable change failed') if status[:enable]
```

Comment:
LGTM.  do you not want to ensure the launchd provider is used in the puppet manifests?